src/passes/RemoveUnusedModuleElements.cpp

```diff
@@ -345,6 +378,36 @@ struct Analyzer {
     }
   }
 
+  void useStructField(StructField structField) {
+    if (!readStructFields.count(structField)) {
+      // Avoid a structured binding as the C++ spec does not allow capturing
+      // them in lambdas, which we need below.
+      auto type = structField.first;
+      auto index = structField.second;
```

Comment:
If we keep the `std::pair`, maybe just have the lambda capture `structField` and use `structField.first` and `structField.second` inside it? That would save having to comment about C++ limitations, at least.

Reply:
I'm not sure how to avoid mentioning the C++ limitation, as the code seems to immediately raise the question "why not structured binding?" E.g.
```diff
diff --git a/src/passes/RemoveUnusedModuleElements.cpp b/src/passes/RemoveUnusedModuleElements.cpp
index c1a47afa9..45225ee1b 100644
--- a/src/passes/RemoveUnusedModuleElements.cpp
+++ b/src/passes/RemoveUnusedModuleElements.cpp
@@ -380,10 +380,7 @@ struct Analyzer {
 
   void useStructField(StructField structField) {
     if (!readStructFields.count(structField)) {
-      // Avoid a structured binding as the C++ spec does not allow capturing
-      // them in lambdas, which we need below.
       auto type = structField.first;
-      auto index = structField.second;
 
       // This is the first time we see a read of this data. Note that it is
       // read, and also all subtypes since we might be reading from them as
@@ -392,6 +389,9 @@ struct Analyzer {
         subTypes = SubTypes(*module);
       }
       subTypes->iterSubTypes(type, [&](HeapType subType, Index depth) {
+        // We cannot use |index| from the outer scope due to C++ limitations on
+        // capturing structured bindings, so only refer to |structField| here.
+        auto index = structField.second;
         auto subStructField = StructField{subType, index};
         readStructFields.insert(subStructField);
 
```